### PR TITLE
Tick consensus while recovering private ledger

### PR DESCRIPTION
**Pull request by jumaffre**
Resolves https://github.com/microsoft/CCF/issues/1213

We now tick the consensus while reading the private ledger. This wasn't the case before, which meant that the last submitted recovery share was not committed until the private ledger was fully recovered.

Also removed the raft suspension mechanism which was used on recovery to make sure that no private transaction was committed while the private ledger was recovered. This is now redundant as the user frontend (the only one that writes private transactions) is closed until the recovery is complete. The `users` and `user_certs` tables are also now public so that all governance tables are public - there was no point keeping them private as the members' proposals to add new users are public anyway!

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -482,10 +482,6 @@ if(BUILD_TESTS)
       # purpose to check that a recovery network is robust to a view change.
       --raft-election-timeout
       4000
-      # Smaller ledger chunks than default to make sure that a ledger made of
-      # multiple chunks can be recovered
-      --ledger-chunk-max-bytes
-      20KB
```
Comment (jumaffre) [inline CMakeLists.txt:488]: All end-to-end tests use this value now, so we test chunking for all tests.

File: src/node/node_state.h [hunk 2]
```diff
@@ -1347,7 +1355,9 @@ namespace ccf
               "Unexpected: write to service table does not include key 0");
           }
 
-          if (it->second.value().status == ServiceStatus::OPEN)
+          if (
+            it->second.value().status == ServiceStatus::OPEN &&
+            this->is_part_of_network())
```
Comment (jumaffre) [inline src/node/node_state.h:1360]: On recovery, we used to open the user frontend way too early when we were replaying the previous service opening.

**Comment by achamayou** [comment]
@jumaffre on the governance_history test, you'll want:

```
    ADDITIONAL_ARGS
      --ledger-chunk-max-bytes
      100Mb
```

Otherwise it chunks and the bit of the logic (circa :66) that assumes it doesn't will fail.

I'd have done it with a suggestion inline, but GitHub doesn't allow it more than a few lines away from an existing change in the PR.

**Comment by jumaffre** [comment]
@achamayou Yup! Done now. It is linked to https://github.com/microsoft/CCF/issues/1275.

**Comment by achamayou** [comment]
@jumaffre it probably makes sense to merge #1308 to unlock the build on this.

**Pull request merged**